query-engine/query-structure/src/field/scalar.rs

```diff
@@ -155,15 +155,22 @@ impl ScalarField {
     }
 
     pub fn native_type(&self) -> Option<NativeTypeInstance> {
-        let (_, name, args, span) = match self.id {
+        let connector = self.dm.schema.connector;
+
+        let raw_nt = match self.id {
             ScalarFieldId::InModel(id) => self.dm.walk(id).raw_native_type(),
             ScalarFieldId::InCompositeType(id) => self.dm.walk(id).raw_native_type(),
-        }?;
-        let connector = self.dm.schema.connector;
+        };
+
+        let psl_nt = raw_nt
+            .and_then(|(_, name, args, span)| connector.parse_native_type(name, args, span, &mut Default::default()));
 
-        let nt = connector
-            .parse_native_type(name, args, span, &mut Default::default())
-            .unwrap();
+        let scalar_type = match self.id {
+            ScalarFieldId::InModel(id) => self.dm.walk(id).scalar_type(),
+            ScalarFieldId::InCompositeType(id) => self.dm.walk(id).scalar_type(),
+        };
+
+        let nt = psl_nt.or_else(|| scalar_type.and_then(|st| connector.default_native_type_for_scalar_type(&st)))?;
```

Comment:
To enable applying special conversion in quaint for specific native types, I updated the `ScalarField::native_type` function to return the default native type in case none are specified. Previously, this function would only return something _if_ a native type was explicitly specified.